src/lib/layer-manager.js

```diff
@@ -138,6 +167,42 @@ export default class LayerManager {
     return this;
   }
 
+  addEventListeners(eventNames) {
```

Comment:
Hm...so then `EventManager` or an equivalent would be passed into `LayerManager`? This feels a bit heavy, though I can see how it may improve extensibility for event handling. I wonder though how useful that flexibility is, given it would slightly increase complexity of the API.

We could make it an optional param and assume most people will use the default, which we would set to `EventManager`. Is that what you're thinking?